lib/internal/modules/esm/translators.js

```diff
@@ -279,7 +279,8 @@ translators.set('commonjs', async function commonjsStrategy(url, source,
   // obtained by calling the monkey-patchable CJS loader.
   const cjsLoader = source == null ? (module, source, url, filename) => {
     try {
-      module.load(filename);
+      assert(module === CJSModule._cache[filename]);
+      CJSModule._load(filename);
```

Comment:
What if a tool is modifying CommonJS modules after they’re loaded, by changing what’s in `require.cache`? I feel like that’s a valid use case, like in mocking or hot module reload; would this change break such behavior?

Reply:
AFAICT it wouldn’t

Reply:
Should we add a test for this?

Reply:
If it's testable, I think that's a good idea :)

Reply:
Agree having a test sounds good, I don't plan on working on this without a repro though, so PR's welcome.

Reply:
It looks like there is one: https://github.com/nodejs/node/issues/49497#issuecomment-1706232114

Reply:
That's a repro of the core issue which there's already a test for. The question was specifically about the behaviour when modifying `require.cache`.

Reply:
@aduh95 and @GeoffreyBooth,

This seems to be breaking Yarn's plug-n-play. I'll see if I can find a simple repo case.

At the moment, all I know is that it works in 20.5.1 and breaks in 20.6.1.

Reply:
Does it work on 20.6.0?

Reply:
I'm fairly confident that https://github.com/nodejs/node/pull/47999 is what is affecting PnP, but it didn't really broke it, did it? It only changed the CJS implementation to a one who is more limited, Yarn maintainers have been informed released a fix before it merged on Node.js side. Upgrading Yarn is the "correct" move forward. Anyway it's off topic, if you see an issue with an up-to-date version of Node.js and Yarn, please open a new issue.

Reply:
@nicolo-ribaudo,

Yes, it worked in 20.6.0 and broke in 20.6.1.